### PR TITLE
SetStats: show stat Tiers above T10 or below T0

### PR DESCRIPTION
**Pull request by robojumper**
Not only is it inconsistent that the UI caps stats at 10 but adds a .5, but LO showed me a 119 recovery set as T10.5 recovery, which is even more wrong.

### CODE REVIEW

**Comment by bhollis** [comment]
When I last tried to do this I was told that people would want to know if they'd gone over T10. Should fix whatever makes 119 say 10.5 instead of 11 though.

**Comment by nev-r** [comment]
11.5 actually :grin:

**Comment by robojumper** [comment]
Updated, also with some tests for negative edge cases.
grafik

**Comment by nev-r** [comment]
in the absence of evidence, which i can collect tomorrow, a list of things to investigate:
- do negative tiers exist in-game?
- what does this display at -10 str?

separately,
i get why `-0.5`, it makes sense as a way to explain "a half mod will fix this"
but it's just as interpretable as "`-0.5` through `-0.9`"

**Comment by robojumper** [comment]
> do negative tiers exist in-game?

No. In-game, stat tiers are clamped to `[0; 10]`. The lowest I can get to is -22 intellect, and it has the same effect as T0 effect, at least according to the tooltip cooldown. You probably have to deliberately try and create negative stat tiers in LO to hit this.

> what does this display at -10 str?

`T-1`. Added this test case.

**Review by nev-r — APPROVED** [review]

**Comment by nev-r** [comment]
i like it

File: src/app/loadout-builder/utils.ts [hunk 1]
```diff
@@ -2,17 +2,27 @@ import { DimItem } from 'app/inventory/item-types';
...
+  const base_tier = Math.floor(stat / 10);
+  // ensure (-3 mod 10) results in a remainder of 7 and a stat of -0.5
+  const point_five = remEuclid(stat, 10) >= 5;
+  const sign = base_tier < 0 ? '-' : '';
+  return `${sign}${base_tier < 0 && point_five ? Math.abs(base_tier + 1) : Math.abs(base_tier)}${
+    point_five ? '.5' : ''
+  }`;
+}
+
+export function remEuclid(dividend: number, divisor: number) {
```
Comment (bhollis) [inline src/app/loadout-builder/utils.ts:24]: This deserves documentation
